### ISSUE TITLE
Activity and Calories

### ISSUE THREAD

**Issue by desmeit** [dariosalvi78/cordova-plugin-health#119]
I store some Data to Health:

```
 navigator.health.store({
                                                             startDate:  new Date(new Date().getTime() - 20 * 60 * 1000), // 20 minutes ago
                                                             endDate: new Date(),
                                                             dataType: 'activity',
                                                             value: 'functional_strength',
                                                             calories: 200,
                                                             sourceName: 'XXX',
                                                             sourceBundleId: 'XXX'
                                                             }, function (data){
                                                             console.log("health store: " + JSON.stringify(data));
                                                             }, function (error){
                                                             console.log("health store error: " + error);
                                                             });
```

It is working perfectly but one question. At the Health activities I can see the calories within the activity but the calories are not added to the Active Calories. What I'm doing wrong? Do I have to save the calories additionally? Or what is the idea behind it?

**Comment by dariosalviwork** [comment]
when you say that 
> the calories are not added to the Active Calories

do you mean that you are retrieving those calories with the plugin?

**Comment by desmeit** [comment]
I mean the app shows the activities AND the active calories. But when I store the activity 'functional_strength' with 200 calories to health, I'm also shown the calories in the activity, but not in the active calories. I had expected that the calories of the activity will be added to the total calories.

![img_0836](https://user-images.githubusercontent.com/9454529/46206487-73923200-c324-11e8-9efd-646c08761b5c.jpg)
![img_0837 2](https://user-images.githubusercontent.com/9454529/46206491-755bf580-c324-11e8-9880-afbc09caa653.jpg)

**Comment by desmeit** [comment]
and in addition I get the following error in xcode:

`*** Terminating app due to uncaught exception '_HKObjectValidationFailureException', reason: 'HKQuantitySample   (2018-09-28 14:13:50 +0200 - 2018-09-28 14:34:50 +0200) requires unit of type Length. Incompatible unit: (null)'`

**Comment by dariosalviwork** [comment]
I see.
I don't know really, it looks like it is how HealthKit works.
What the plugin does is simply storing a workout. [This](https://developer.apple.com/documentation/healthkit/hkworkout) is the official documentation.

For what regards the error code, when is it launched? When you store the activity?
Is the distance stored with the activity correctly?

**Comment by desmeit** [comment]
Okay, regarding the error code: it seems the distance is required. but why do I need a distance for a functional_strength workout?🤔

Regarding the calories: I should store the active calories separately, right?

Still one note: If I request a authorization for "workouts" instead of "activity" it is not possible to store with dataType: 'workouts'. dataType: 'activity' is working. is that right?

Thanks

**Comment by dariosalviwork** [comment]
Distance: in principle it's optional, but there might be a bug.

Calories: I don't know what the answer is really. From your experiments it seems that, yes, if you want those calories to appear as active calories, you need to add them manually. BTW, this happens in iOS, probably, in Google Fit it would be different.

Permissions: correct, at the moment the _workout_ datatype is only for the permissions phase, then you should use _activity_ in your store() and query().

**Comment by desmeit** [comment]
Thanks for the explaination.
Still one question: Storing is working but I get the followging error (although I confirmed the request):

Authorization is not determined

**Comment by dariosalviwork** [comment]
What do you mean with "is working" but "I get an error" ? Is it working or not?

**Comment by desmeit** [comment]
I get this error through errorCallback but it is storing in health.

**Comment by dariosalviwork** [comment]
Mmmm, maybe it has to do with the storing of calories and/or distance.
You may need permission for those too?

**Comment by desmeit** [comment]
That was the reason. Thanks a lot! I think it would be nice to make the distance optional, but I like this plugin a lot.

**Comment by corysmc** [comment]
@dariosalviwork Is there a related issue logged somewhere else for the distance required to log calories? I have the same issue: When trying to log an activity with calories, it won't store the calories successfully without having `distance:0` added (which also requires distance permissions - which I shouldn't need).

**Comment by dariosalviwork** [comment]
yes, and there's an easy and general fix for it (as pointed out in issue 120). Any good soul who wants to  fix it for everybody?
